### PR TITLE
Add music protobuf and BUILD file.

### PR DESCRIPTION
**Pull request by adarob**

### CODE REVIEW

**Comment by fredbertsch** [comment]
Add the license text to the build and proto files. (This is tricky on my phone, so I'll look closer tomorrow.)

File: magenta/protobuf/music.proto [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+package tensorflow.magenta;
+
+// A message containing a symbolic music sequence. The design is largely
+// based on MIDI but it should be able to represent any music sequence.
+// For details see https://www.midi.org/specifications.
+message NoteSequence {
+  // NOTE: The first three fields are duplicated from SequenceFile to allow
+  // creation of portable SSTables of Sequence protos.
+  // Unique id. See data_utils.h GenerateSequenceId().
```
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: Last sentence isn't help.

File: magenta/protobuf/music.proto [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  string id = 1;
+  // The path of the file relative to the root of the collection.
+  string filename = 2;
+  // The collection from which the file comes. This can be shorthand e.g.
+  // "bach". One purpose is to allow for easy selection of all or some files
+  // from a particular source.
+  string collection = 3;
```
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: Presumably this is not a globally unique id. Unless that's the case, let's call it "collection_name" to make it sound less like a centrally managed name.

File: magenta/protobuf/music.proto [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+  // MIDI ticks per beat, also known as resolution or PPQ ("parts per quarter").
+  // There is no widely-used default. 220 is assumed per the choice made in
+  // third_party/py/pretty_midi.
```
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: The last sentence needs to be rewritten. "If unspecified, 220 is assumed."

Not that this is the right place to complain, but 220 is a weird default value. Why not make triplets work perfectly? I'd vote for 60.

File: magenta/protobuf/music.proto [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      B = 11;
+    }
+
+    enum Mode {
+      MAJOR = 0;
+      MINOR = 1;
+    }
+  }
+
+  message TempoChange {
```
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: The first one isn't a TempoChange, it's a Tempo. Let's call them all Tempos.

File: magenta/protobuf/music.proto [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    // Time in seconds.
+    double time = 1;
+    // Tempo in beats per minute.
+    double bpm = 2;
+  }
+
+  // Stores MIDI PitchBend data. See the MIDI specfication for details.
```
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: Can you link to the portion of the MIDI spec? There's a bunch of oddness in this message that'd be useful to look up.

File: magenta/protobuf/music.proto (hunk 3 above)
Comment (lasconic) [inline magenta/protobuf/music.proto]: FWI, [MuseScore](https://musescore.org) uses 480 and quintuplets work perfectly too.

File: magenta/protobuf/music.proto (hunk 1 above)
Comment (adarob) [inline magenta/protobuf/music.proto]: Removed this and other internal references.

File: magenta/protobuf/music.proto (hunk 2 above)
Comment (adarob) [inline magenta/protobuf/music.proto]: Are you referring to the id? It is intended to be globally unique.

File: magenta/protobuf/music.proto (hunk 3 above)
Comment (adarob) [inline magenta/protobuf/music.proto]: @douglaseck thoughts?

File: magenta/protobuf/music.proto (hunk 4 above)
Comment (adarob) [inline magenta/protobuf/music.proto]: Fixed.

File: magenta/protobuf/music.proto (hunk 5 above)
Comment (adarob) [inline magenta/protobuf/music.proto]: Unfortunately the spec is in a non-linkable format.

File: magenta/protobuf/music.proto (hunk 2 above)
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: I was talking about "collection". Presumably, there are a finite number of interesting MIDI collections out there, but, once we're in OSS land, there's no central naming authority for them. So, I don't think they will be unique. (Unless I'm missing something?) You don't say anything about this being either unique or human readable. A comment about that would be helpful. If it's not unique, then a name that emphasizes the field's purpose would also be helpful.

File: magenta/protobuf/music.proto (hunk 3 above)
Comment (fredbertsch) [inline magenta/protobuf/music.proto]: Looks like @douglaseck is okay with this. (Offline discussion)

**Pull request merged**